frontend/src/pages/AccountSend.vue

```diff
@@ -741,6 +742,25 @@ function useSendForm() {
 
       const validAmount = Boolean(humanAmountValue) && isValidTokenAmount(humanAmountValue as string) === true;
       isValidForm.value = isValidRecipientId.value && Boolean(tokenValue) && validAmount;
+
+      let validatedBatchSendForm = true;
+      const isValidRecipientPromises: Promise<boolean | string>[] = [];
+      for (const batchSend of batchSends.value) {
+        if (validatedBatchSendForm) {
+          const { token, amount, receiver } = batchSend;
+          const isValidAmount = Boolean(amount) && isValidTokenAmount(amount, token);
+          const isValidToken = Boolean(token);
+          isValidRecipientPromises.push(isValidId(receiver));
+          if (isValidAmount !== true || !receiver || !isValidToken) validatedBatchSendForm = false;
+        }
+      }
+      await Promise.all(isValidRecipientPromises).then((results) => {
```

Comment:
Small optimization, if `validatedBatchSendForm` is false we can skip this.

```
if (validatedBatchSendForm) {
  await Promise.all(...)
}

```

Reply:
Thanks! https://github.com/ScopeLift/umbra-protocol/pull/538/commits/fa463d673f6f8388afbc7847637c8d71718084b3